File: abc/abc321/b.cpp
// https://atcoder.jp/contests/abc321/submissions/45825518

#include <bits/stdc++.h>
using namespace std;

int main() {
  int n, x;
  cin >> n >> x;

  vector<int> a(n - 1);
  for(auto &z : a) cin >> z;
  sort(a.begin(), a.end());

  for(int i = 0; i <= 100; i++) {
    vector<int> b;
    for(auto x : a) b.push_back(x);
    b.push_back(i);
    sort(b.begin(), b.end());

    int sum = 0;
    for(int j = 1; j < b.size() - 1; j++) {
      sum += b[j];
    }

    if(sum >= x) {
      cout << i;
      return 0;
    }
  }

  cout << -1;
}

File: abc/abc321/b.java
// https://atcoder.jp/contests/abc321/submissions/45898429

import java.util.Scanner;
import java.util.Arrays;

class Main {
  public static void main(String args[]) {
    Scanner stdIn = new Scanner(System.in);

    int n = stdIn.nextInt();
    int x = stdIn.nextInt();

    int[] a = new int[n - 1];
    for(int i = 0; i < n - 1; i++) {
      a[i] = stdIn.nextInt();
    }

    for(int i = 0; i <= 100; i++) {
      int[] b = new int[n];
      for(int j = 0; j < n - 1; j++) {
        b[j] = a[j];
      }
      b[n - 1] = i;
      Arrays.sort(b);

      int sum = 0;
      for(int j = 1; j < n - 1; j++) {
        sum += b[j];
      }

      if(sum >= x) {
        System.out.println(i);
        System.exit(0);
      }
    }

    System.out.println(-1);
  }
}
